### PR TITLE
[RLlib] Support lowercase Torch activation aliases

### LINKED ISSUE #61076: [Core] Is there a reason LeakyReLU activations aren't supported by `get_activation_fn`?
### Description

Given the breadth of ML tasks Ray covers, including LLM fine-tuning and reinforcement learning among others, there's a surprisingly small set of allowed activation functions in [utils.get_activation_fn](https://github.com/ray-project/ray/blob/master/rllib/core/models/torch/primitives.py#L15). 

```
if name_lower in ["linear", None]:
            return None

        _, nn = try_import_torch()
        # First try getting the correct activation function from nn directly.
        # Note that torch activation functions are not all lower case.
        fn = getattr(nn, name, None)
        if fn is not None:
            return fn

        if name_lower in ["swish", "silu"]:
            return nn.SiLU
        elif name_lower == "relu":
            return nn.ReLU
        elif name_lower == "tanh":
            return nn.Tanh
        elif name_lower == "elu":
            return nn.ELU
        elif name_lower == "softmax":
            return nn.Softmax
```

GeLU seems pretty widely used as a drop-in ReLU replacement for modern transformers, and Leaky ReLU [does a much better job of improving plasticity in nonstationary tasks](https://arxiv.org/html/2509.22562v1), which seems pertinent to RLlib seeing as how value head optimization, among other things, is inherently nonstationary. 

Come to think of it, is there any reason we can't just drop in the full list of [PyTorch-supported activation functions](https://docs.pytorch.org/docs/main/nn.html) as a dictionary mapping lowercase names to functions? There's no special logic being handled here.

### Use case

Better performance on RL tasks, especially as it relates to network plasticity. More flexibility in hyperparameter search, likely many other benefits for other users.

### PR DESCRIPTION
**Pull request by HasanTasdiq**
Fixes #61076

## Summary
- Added lowercase Torch activation aliases for `gelu`, `leaky_relu`, and `leakyrelu` in `get_activation_fn`.
- Added focused unit coverage for alias resolution without requiring a real Torch install.

## Why
RLlib config users can pass lowercase activation names such as `relu` and `silu`, but lowercase GELU and LeakyReLU names were not resolved for Torch even though the canonical class names worked.

## Testing
- Ran: `.venv/bin/python -m pytest -v -s rllib/models/tests/test_utils.py` — passed
- Ran: `.venv/bin/ruff check rllib/models/utils.py rllib/models/tests/test_utils.py` — passed
- Ran: `.venv/bin/black --check rllib/models/utils.py rllib/models/tests/test_utils.py` — passed
- Ran: `.venv/bin/python -m py_compile rllib/models/utils.py rllib/models/tests/test_utils.py` — passed

## Notes
- `.venv/bin/pre-commit run --files rllib/models/utils.py rllib/models/tests/test_utils.py` could not complete because `nodeenv` failed SSL certificate verification while setting up an unrelated Node environment. The relevant Python lint checks passed directly.

### CODE REVIEW

File: rllib/models/utils.py [hunk 1]
```diff
@@ -12,7 +12,7 @@ def get_activation_fn(
     """Returns a framework specific activation function, given a name string.
 
     Args:
-        name: One of "relu" (default), "tanh", "elu",
+        name: One of "relu" (default), "tanh", "elu", "gelu", "leaky_relu",
```
[the remark this answers is not in the document]
Comment (HasanTasdiq) [reply rllib/models/utils.py:15]: Updated the docstring to list the existing `leakyrelu` and `softmax` aliases as well. Thanks for catching the mismatch.

**Comment by pseudo-rnd-thoughts** [comment]
Thanks for the PR but beyond the most basic of activation functions then I think users should pass the direct function (nn.LeakyReLU) rather than a string to ensure that the function exists so I'm going to close this.
